[DOCS] Rewrite "What is Elasticsearch?" (Part 1) (#112213)
elastic/elasticsearch: docs/reference/intro.asciidoc

{es} is used for a wide and growing range of use cases. Here are a few examples:

* *Monitor log and event data*. Store logs, metrics, and event data for observability and security information and event management (SIEM).
* *Build search applications*. Add search capabilities to apps or websites, or build enterprise search engines over your organization's internal data sources.
* *Vector database*. Store and search vectorized data, and create vector embeddings with built-in and third-party natural language processing (NLP) models.
* *Retrieval augmented generation (RAG)*. Use {es} as a retrieval engine to augment Generative AI models.
* *Application and security monitoring*. Monitor and analyze application performance and security data effectively.
* *Machine learning*. Use {ml} to automatically model the behavior of your data in real-time.

This is just a sample of search, observability, and security use cases enabled by {es}. 

> Refer to our https://www.elastic.co/customers/success-stories[customer success stories] for concrete examples across a range of industry verticals.

Reviewer: ```suggestion
Refer to our https://www.elastic.co/customers/success-stories[customer success stories] for concrete examples across a range of industries.
```

// Link to demos, search labs chatbots

[discrete]
[[elasticsearch-intro-elastic-stack]]
.What is the Elastic Stack?
*******************************
{es} is the core component of the Elastic Stack, a suite of products for collecting, storing, searching, and visualizing data.
https://www.elastic.co/guide/en/starting-with-the-elasticsearch-platform-and-its-solutions/current/stack-components.html[Learn more about the Elastic Stack].
*******************************
// TODO: Remove once we've moved Stack Overview to a subpage?

[discrete]
[[elasticsearch-intro-deploy]]
=== Deployment options

To use {es}, you need a running instance of the {es} service.
You can deploy {es} in various ways:

* <<run-elasticsearch-locally,*Local dev*>>. Get started quickly with a minimal local Docker setup. 
* {cloud}/ec-getting-started-trial.html[*Elastic Cloud*]. {es} is available as part of our hosted Elastic Stack offering, deployed in the cloud with your provider of choice. Sign up for a https://cloud.elastic.co/registration[14 day free trial].

> * {serverless-docs}/general/sign-up-trial[*Elastic Cloud Serverless* (technical preview)]. Create serverless projects for autoscaled and fully-managed {es} deployments. Sign up for a https://cloud.elastic.co/serverless-registration[14 day free trial].

Reviewer: ```suggestion
* {serverless-docs}/general/sign-up-trial[*Elastic Cloud Serverless* (technical preview)]. Create serverless projects for autoscaled and fully managed {es} deployments. Sign up for a https://cloud.elastic.co/serverless-registration[14 day free trial].
```

**Advanced deployment options**

> * <<elasticsearch-deployment-options,*Self managed*>>. Install, configure, and run {es} on your own premises.

Reviewer: ```suggestion
* <<elasticsearch-deployment-options,*Self-managed*>>. Install, configure, and run {es} on your own premises.
```

Reviewer: no more hyphen nits, I promise! I started seeing them and couldn't stop 👀

[[elasticsearch-intro]]
== What is {es}?

> https://github.com/elastic/elasticsearch[{es}] is a distributed, RESTful search and analytics engine, scalable data store, and vector database built in Java on top of the Apache Lucene library.

Author: Lucene is under the hood, but important for search practitioners so keeping that.

> might be nice because being able to handle huge workloads is something we used to cover but removed from this page

👍

> https://github.com/elastic/elasticsearch[{es}] is a distributed, RESTful search and analytics engine, scalable data store, and vector database built in Java on top of the Apache Lucene library.

Author: > built in Java (irrelevant to anyone but contributors)

Not necessarily. People like to know what stuff is written in. Like sometimes someone will see something is written in Rust and say "Hey this is my kind of project".

> Use {es} to search, index, store, and analyze data of all shapes and sizes in near real-time.

Author: nah just real-time bad grammar from me ;)

[TIP]
====
{es} has a lot of features. Explore the full list on the https://www.elastic.co/elasticsearch/features[product webpage^].
====

> {es} is the heart of the the <<elasticsearch-intro-elastic-stack,{stack}>> and powers the Elastic https://www.elastic.co/enterprise-search[Search], https://www.elastic.co/observability[Observability] and https://www.elastic.co/security[Security] solutions.

Author: classic

{es} is used for a wide and growing range of use cases. Here are a few examples:

> * *Monitor log and event data*. Store and analyze logs, metrics, and security event data for operational insights and SIEM.

Author: Agree this needs rephrasing: " Store logs, metrics, and event data for observability and security information and event management (SIEM)."

* *Build search applications*. Add search capabilities to apps or websites and build enterprise search engines over your organization's internal data sources.
* *Vector database*. Store and search vectorized data, create vector embeddings with built-in and third-party NLP models.
* *Retrieval augmented generation (RAG)*. Use {es} as a retrieval engine to augment Generative AI models.
* *Application and security monitoring*. Monitor and analyze application performance and security data effectively.
* *Machine learning*. Use {ml} to automatically model the behavior of your data in real-time.

This is just a sample of search, observability, and security use cases enabled by {es}. 
Refer to our https://www.elastic.co/customers/success-stories[customer success stories] for concrete examples across a range of industry verticals.
// Link to demos, search labs chatbots

[discrete]
[[elasticsearch-intro-elastic-stack]]
.What is the Elastic Stack?
*******************************

> The Elastic Stack refers to the suite of products enabled by {es}:

Author: No I agree enabled isn't right

* https://www.elastic.co/guide/en/kibana/current/index.html[Kibana]. A UI for visualizing and exploring data in {es}.
* https://www.elastic.co/guide/en/elasticsearch/client/index.html[Client libraries]. Work with {es} in your preferred programming language.
* https://www.elastic.co/guide/en/logstash/current/introduction.html[Logstash]. A server-side data processing pipeline for ingesting and transforming data from multiple sources and indexing into {es}.
* https://www.elastic.co/guide/en/fleet/current/fleet-overview.html[Fleet and Elastic Agent.] Elastic Agents is a single, unified way to add monitoring for logs, metrics, and other types of data to a host. Fleet is a central place to configure and monitor your Elastic Agents.
* https://www.elastic.co/guide/en/beats/libbeat/current/beats-reference.html[Beats]. Lightweight data shippers for sending data from edge machines to {es}.
* https://www.elastic.co/guide/en/observability/current/apm.html[APM]. Monitor the performance of your applications.

> * https://www.elastic.co/guide/en/elasticsearch/hadoop/current/float.html[{es} Hadoop]. Use {es} as a Hadoop input/output format.

Author: agree, will remove

https://www.elastic.co/guide/en/starting-with-the-elasticsearch-platform-and-its-solutions/current/stack-components.html[Learn more about the Elastic Stack].
*******************************
// TODO: Remove once we've moved Stack Overview to a subpage?

[discrete]
[[elasticsearch-intro-deploy]]
=== Deployment options

To use {es}, you need a running instance of the {es} service.

> You can deploy {es} in various ways:

Reviewer 2: I think we clearer "some of these cost money and some of these are free" messaging. ECE and ECK are especially confusing in comparison to self-managed. can also be a follow-up

Author: Maybe we could categorize self-managed, ECE, and ECK into an "advanced deployment" options section

Reviewer 2: that would work for me, although self-managed is also the beginner/builder path

Author: that's why local dev has its own category and has top billing 😄

Reviewer 2: ah my bad

* https://elastic.co/guide/en/cloud/current/ec-getting-started.html[*Elastic Cloud*]. {es} is available as part of our hosted Elastic Stack offering, deployed in the cloud with your provider of choice. Sign up for a https://cloud.elastic.co/registration[14 day free trial].
* https://elastic.co/guide/en/cloud-enterprise/current/Elastic-Cloud-Enterprise-overview.html[*Elastic Cloud Enterprise*]. Deploy Elastic Cloud on public or private clouds, virtual machines, or your own premises.
* https://elastic.co/guide/en/cloud-on-k8s/current/k8s-overview.html[*Elastic Cloud on Kubernetes*]. Deploy Elastic Cloud on Kubernetes.
* https://www.elastic.co/docs/current/serverless[*Elastic Cloud Serverless* (technical preview)]. Create serverless projects for autoscaled and fully-managed {es} deployments. Sign up for a https://cloud.elastic.co/serverless-registration[14 day free trial].
* <<elasticsearch-deployment-options,*Self managed*>>. Install, configure, and run {es} on your own premises.
+
[TIP]
====
If you just want to get started quickly with a minimal local setup, refer to <<run-elasticsearch-locally,Run {es} locally>>.

> ====

Author: Agree!

[discrete]
[[elasticsearch-next-steps]]
=== Learn more

> * <<getting-started, Quickstart>>. A beginner's guide to deploying your first {es} instance, indexing data, and running queries.

Reviewer 2: needs a little glue
```suggestion

Review the following resources to get started with {es}:

* <<getting-started, Quickstart>>. A beginner's guide to deploying your first {es} instance, indexing data, and running queries.
```

Author: Glue FTW. Little less formal: "Here are some resources to help you get started:"

Reviewer 2: I've always avoided "there are/is" and "here are/is" in my own writing but I'm cool with that

used at search time. When you query a full-text field, the query text undergoes
the same analysis before the terms are looked up in the index.

[[search-analyze]]

> === Information out: search and analyze

Reviewer 2: afraid to lose this search overview here. I don't think it's good but I don't think we're adequately pointing to a replacement ATM

Author: yeah we have no backlinks in here though so I was happy to cut it out and do a full replacement later but I can leave it in

that match your users' search criteria--for example, all size 70 _non-stick
embroidery_ needles.

[[scalability]]

> === Scalability and resilience

Reviewer 2: these headings are so much better ty

Reviewer: 💯

the active leader index and handles all write requests. Indices replicated to
secondary clusters are read-only followers.

[discrete]
[[admin]]

> ==== Care and feeding

Reviewer 2: while I hate this heading, I do think this is its own concept

Author: ok I'll rename it
